vendor/github.com/waku-org/go-waku/waku/v2/peermanager/peer_manager.go

```diff
@@ -433,6 +433,11 @@ func (pm *PeerManager) AddDiscoveredPeer(p service.PeerData, connectNow bool) {
 	//Check if the peer is already present, if so skip adding
 	_, err := pm.host.Peerstore().(wps.WakuPeerstore).Origin(p.AddrInfo.ID)
 	if err == nil {
+		//Add addresses if existing addresses have expired
+		existingAddrs := pm.host.Peerstore().Addrs(p.AddrInfo.ID)
+		if len(existingAddrs) == 0 {
```

Comment:
It is possible, but that means there are addresses for the peer in the peerStore right unless i am missing something.

Reply:
> I'm thinking the addresses in peerStore maybe outdated, do we have logic to update the addresses in such case? It seems not covered by this line of code.

Good Point,  we need to update the addresses as well whenever there is an ENR change because it is possible that node has got an address update. Will check and implement it if not.

Reply:
Have verified in the code and this is already handled at https://github.com/waku-org/go-waku/blob/795322a19665ac48ca1f5cf725f64b1fd1e67808/waku/v2/peermanager/peer_manager.go#L490-L496 which is invoked if new ENR seq is greater than older one.